olad/Systemd.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Systemd.cpp
+ * Provides wrapped access to systemd interfaces.
+ * Copyright (C) 2018 Shenghao Yang
+ */
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif  // HAVE_CONFIG_H
+
+#if HAVE_LIBSYSTEMD
+#include <systemd/sd-daemon.h>
+#endif  // HAVE_LIBSYSTEMD
+
+#include "ola/Logging.h"
+#include "ola/base/Strerror_r.h"
+
+#include "olad/Systemd.h"
+
+namespace ola {
+
+int SystemdNotify(int unset_environment, const char *state) {
+  int rtn = sd_notify(unset_environment, state);
+  if (rtn < 0) {
+    char buf[1024];
+    if (ola::Strerror_r(-rtn, buf, sizeof(buf))) {
+      OLA_WARN << "Error sending notification to systemd: errno = " << -rtn;
```

Comment:
You've now just had to duplicate the "Error sending notification to systemd" text. I still feel the whole return a string or return "errno" and a value should be in the function, as we'll always want that functionality. Or if you want to do a direct wrap of strerror_r, we should also ship a helper function which does what we actually want each time (and then calls the wrapped strerror_r).

Otherwise we have to keep writing this code:
```
    char buf[1024];
    if (ola::Strerror_r(-rtn, buf, sizeof(buf))) {
      OLA_WARN << "Error sending notification to systemd: errno = " << -rtn;
    } else {
      OLA_WARN << "Error sending notification to systemd: " << buf;
    }
```

Whereas we should just be able to write something like this (or whatever the function gets called) and get the same output as the above:
`    OLA_WARN << "Error sending notification to systemd: " << ola::StrError_rHelper(-rtn);`